security/user-last-seen.php

```diff
@@ -0,0 +1,60 @@
+<?php
+
+const LAST_SEEN_UPDATE_USER_META_CACHE_TTL = 60; // Store last seen once per minute to avoid too much write DB operations
+const LAST_SEEN_UPDATE_USER_META_CACHE_KEY_PREFIX = 'vip_last_seen_update_user_meta_cache_key';
+
+add_action( 'set_current_user', 'update_user_last_seen', 10, 0 );
+add_filter('manage_users_columns', 'users_columns' ) ;
+add_filter('manage_users_custom_column', 'users_custom_column', 10, 3 ) ;
+
+function update_user_last_seen() {
+	global $current_user;
+
+	$cache_key = LAST_SEEN_UPDATE_USER_META_CACHE_KEY_PREFIX . $current_user->ID;
```

Comment:
Not 100% sure, but might be possible for the current user ID to be `0` if some code does `wp_set_current_user( 0 );`.  So might want to safeguard that and make sure we actually have an ID first, and that `$current_user` is set / not null / has that ID property.